armory/utils/metrics.py

```diff
@@ -168,30 +172,23 @@ def resource_context(name="Name", profiler=None):
         )
         pr = cProfile.Profile()
         pr.enable()
-    if profiler == "Basic" or profiler == "Deterministic":
-        startTime = time.perf_counter()
+    startTime = time.perf_counter()
     yield
-    if profiler == "Basic" or profiler == "Deterministic":
-        elapsedTime = time.perf_counter() - startTime
-        if profiler == "Deterministic":
-            pr.disable()
-            s = io.StringIO()
-            sortby = "cumulative"
-            ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
-            ps.print_stats()
-            stats = s.getvalue()
-        if name in computational_resource_dict:
-            computational_resource_dict[name]["execution_count"] += 1
-            computational_resource_dict[name]["total_time"] += elapsedTime
-            if profiler == "Deterministic":
-                computational_resource_dict[name]["stats"] += stats
-
-        else:
-            computational_resource_dict[name] = {}
-            computational_resource_dict[name]["execution_count"] = 1
-            computational_resource_dict[name]["total_time"] = elapsedTime
-            if profiler == "Deterministic":
-                computational_resource_dict[name]["stats"] = stats
+    elapsedTime = time.perf_counter() - startTime
+    if profiler == "Deterministic":
+        pr.disable()
+        s = io.StringIO()
+        sortby = "cumulative"
+        ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+        ps.print_stats()
+        stats = s.getvalue()
+    if name not in computational_resource_dict:
+        computational_resource_dict[name] = defaultdict(lambda: 0)
```

Comment:
It looks like if the profiler is "Deterministic", the value for the key "stats" is a string, so that the defaultdict needs to be initialized accordingly